ignite/engine/deterministic.py

```diff
@@ -77,14 +77,14 @@ def setup_batch_indices(self) -> None:
 
     def __iter__(self) -> Generator:
         self.setup_batch_indices()
-        for batch in self.batch_indices:
+        for batch in cast(List, self.batch_indices):
```

Comment:
Can't we annotate `self.batch_indices = []  # type: List` instead of the cast here ?

Reply:
actually the problem is the `None` assignment in the `__init__` function
https://github.com/pytorch/ignite/blob/b2e1b33c2154483ea5204bc1322c8af33e51c0c1/ignite/engine/deterministic.py#L60-L67
therefore `mypy` thinks, it could be `None` at that point.

Reply:
I see, let's set `self.batch_indices = []` in the constructor. Shouldn't be that different though.